server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -548,11 +549,16 @@ protected Node(
              */
             this.environment = new Environment(settings, initialEnvironment.configDir(), Node.NODE_LOCAL_STORAGE_SETTING.get(settings));
             Environment.assertEquivalent(initialEnvironment, this.environment);
-            if (DiscoveryNode.isSearchNode(settings) == false) {
-                nodeEnvironment = new NodeEnvironment(tmpSettings, environment);
-            } else {
-                nodeEnvironment = new NodeEnvironment(settings, environment, new FileCacheCleaner(this::fileCache));
-            }
+            IndexStoreListener.CompositeIndexStoreListener compositeIndexStoreListener = new IndexStoreListener.CompositeIndexStoreListener(
+                pluginsService.filterPlugins(IndexStorePlugin.class)
+                    .stream()
+                    .map(IndexStorePlugin::getIndexStoreListener)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .collect(Collectors.toList())
+            );
+            compositeIndexStoreListener.addListener(new FileCacheCleaner(this::fileCache));
```

Comment:
I was actually debating between `default Optional<IndexStoreListener> getIndexStoreListener()` and `default List<IndexStoreListener> getIndexStoreListener()` in `IndexStorePlugin` interface. I went with the former because I didn't think there would be many implementations of this class but I'm not sure if that's actually a good reason to do that.

Reply:
I think `default Optional<IndexStoreListener> getIndexStoreListener()` is a right choice and consistent with other hook APIs we use for listener. The `Stream.concat` comment was regarding replacing `addListener` with constructor based initialization.

Reply:
Thanks for the input! Yeah I realize my comment wasn't directly related to the `Stream.concat` point but it felt like the best place to ask the question 😅